src/rustfmt_diff.rs

```diff
@@ -88,6 +101,7 @@ pub fn print_diff<F>(diff: Vec<Mismatch>, get_section_title: F)
     where F: Fn(u32) -> String
 {
     let mut t = term::stdout().unwrap();
+    let isatty = unsafe { libc::isatty(libc::STDOUT_FILENO) } != 0;
```

Comment:
fwiw, rustc does this for unix: https://github.com/rust-lang/rust/blob/f12d32d307d53a8f7851ce8dd1b4fbd7a4f0b50a/src/libsyntax/errors/emitter.rs#L659